substrate/client/transaction-pool/tests/zombienet/yap.rs

```diff
@@ -0,0 +1,162 @@
+use std::{path::PathBuf, time::SystemTime};
+
+use zombienet_sdk::{
+	LocalFileSystem, Network as ZNetwork, NetworkConfig, NetworkConfigBuilder, NetworkConfigExt,
+};
+
+use super::{
+	Network, Node, ParachainConfig, RelaychainConfig, DEFAULT_BASE_DIR, DEFAULT_PC_NODE_RPC_PORT,
+	DEFAULT_RC_NODE_RPC_PORT,
+};
+use anyhow::anyhow;
+use which::which;
+
+// A zombienet network with two relaychain 'polkadot' validators and one parachain
+// validator based on yap-westend-live-2022 chain spec.
+pub struct YapNetwork {
+	rc_config: RelaychainConfig,
+	pc_config: ParachainConfig,
+	rc_nodes: Vec<Node>,
+	pc_nodes: Vec<Node>,
+	base_dir: PathBuf,
+}
+
+impl YapNetwork {
+	/// Creates a new [`SmallNetworkYap`].
+	pub fn new(
+		rc_config: RelaychainConfig,
+		pc_config: ParachainConfig,
+	) -> Result<Self, anyhow::Error> {
+		// Create temporary network base dir.
+		let datetime: chrono::DateTime<chrono::Local> = SystemTime::now().into();
+		let base_dir =
+			PathBuf::from(format!("{}-{}", DEFAULT_BASE_DIR, datetime.format("%Y%m%d_%H%M%S")));
+		std::fs::create_dir(base_dir.clone()).map_err(|err| anyhow!(format!("{err}")))?;
+
+		Ok(YapNetwork {
+			rc_config,
+			pc_config,
+			rc_nodes: vec![
+				Node::new("alice".to_owned(), vec![], true),
+				Node::new("bob".to_owned(), vec![], true),
+			],
+			pc_nodes: vec![Node::new(
+				"charlie".to_owned(),
+				vec![
+					("--pool-limit", "500000").into(),
+					("--pool-kbytes", "2048000").into(),
+					("--rpc-max-connections", "15000").into(),
+					("--rpc-max-response-size", "150").into(),
+					"-lbasic-authorship=info".into(),
+					"-ltxpool=debug".into(),
+					"-lsync=info".into(),
+					"-laura::cumulus=info".into(),
+					"-lpeerset=info".into(),
+					"-lsub-libp2p=info".into(),
+					"--pool-type=fork-aware".into(),
+					"--experimental-use-slot-based".into(),
+					"--state-pruning=1024".into(),
+					"--rpc-max-subscriptions-per-connection=128000".into(),
+					"--max-runtime-instances=32".into(),
+				],
+				true,
+			)],
+			base_dir,
+		})
+	}
+}
+
+#[async_trait::async_trait]
+impl Network for YapNetwork {
+	fn ensure_bins_on_path(&self) -> bool {
+		// We need polkadot, polkadot-parachain, polkadot-execute-worker, polkadot-prepare-worker,
+		// (and ttxt? - maybe not for the network, but for the tests, definitely)
```

Comment:
Would not say that ttxt would be needed here. Having purely-zombienet oriented abstractions sounds good. We could check ttxt in `TxSpammer` abstraction (which in future could just use `lib` for sending).